embassy-net-nrf91/src/context.rs

```diff
@@ -292,15 +305,34 @@ impl<'a> Control<'a> {
     }
 
     async fn wait_attached(&self) -> Result<Status, Error> {
-        while !self.attached().await? {
-            Timer::after(Duration::from_secs(1)).await;
+        trace!("waiting for attached");
+        // Poll every second, up to 60 seconds, then error out
+        const POLL_INTERVAL: Duration = Duration::from_secs(1);
+        const MAX_TIMEOUT: Duration = Duration::from_secs(60);
```

Comment:
This should be configurable, maybe via some kind of config type passed when creating the context. Because in my case it can take longer than 60 secs, and I've seen some cases where it could take up to 20 minutes to attach in testing (probably some kind of rate limiting if you attach too frequently).